modules/net-cloudnat/variables.tf

```
@@ -44,11 +44,26 @@ variable "config_source_subnetworks" {
     subnetworks = optional(list(object({
       self_link        = string
       all_ranges       = optional(bool, true)
+      primary_range    = optional(bool)
```

```suggestion
      primary_range    = optional(bool, false)
```
And then use bool expressions where checking for `primary_range` above?